Reader Full Post Refresh Design Review

nbradbury opened the pull request:
Fixes #4581 - this is the master branch of the reader "full post refresh." All code related to this feature has already been reviewed and merged into this branch. This PR is to seek a design review of this feature before it's merged into `develop`.

Changes include:

- A redesigned header for the post which includes the blavatar+avatar (or just avatar if there is no blavatar)
- A shortened dateline which includes both the primary and secondary tags
- The like & comment icons in the footer now include text
- A redesign of the related posts section, which now has a "More in [site]" section and a "More on WordPress.com" section


![1](https://cloud.githubusercontent.com/assets/3903757/19571071/a65b547e-96ca-11e6-99ef-0b5048d70aa7.png)

![2](https://cloud.githubusercontent.com/assets/3903757/19571077/b0305620-96ca-11e6-9141-47236f88026e.png)

fraying commented:
Great work! Looks great to me. One question: Will the stories in "MORE ON  WORDPRESS.COM" include images if the stories have them?

nbradbury commented:
> Will the stories in "MORE ON WORDPRESS.COM" include images if the stories have them?

Yes, it will. Here's an example:

![3](https://cloud.githubusercontent.com/assets/3903757/19571496/5c3d011a-96cc-11e6-89f5-bf405014e3c5.png)

fraying commented:
Excellent! I love it. Great work!

nbradbury commented:
A few points I want to mention before the design review gets underway:

- Unlike the web reader, related posts on Android doesn't hyperlink the two "MORE" labels, nor can you tap the header above each related wp.com post to view the site. This is because these are very small tap targets on mobile, so the user could easily tap the header when they meant to tap the post itself.
- Similarly, the follow icons under wp.com related posts doesn't include the follow/following text for the same reason - it would be too easy to tap the text when you intended to tap the post iself.

fraying commented:
Makes sense. I think this is good to go. Any comments, @jan?

nbradbury commented:
> Any comments, @jan?

I think you meant @jancavan :)

fraying commented:
I did. Sorry. I blame the cold meds. 😷

roundhill commented:
Stellar work @nbradbury! The 'Following' text looks a bit misaligned to me, I wonder if we should align it more to the box created by the blavatar and text:

<img width="402" alt="screen shot 2016-10-20 at 11 07 55 am" src="https://cloud.githubusercontent.com/assets/789137/19572082/a3ea301c-96b5-11e6-8e8a-8ff909bfc465.png">

nbradbury commented:
> The 'Following' text looks a bit misaligned to me

Hmm, good eyes as usual @roundhill! That's going be ugly to fix since the follow button is a custom widget that defines its own padding, but it does need to be addressed. I'll take a crack at it.

jancavan commented:
Thanks for getting this done really quickly @nbradbury! 

Wasn't sure why I kept getting a build error, but upgrading to the new version actually fixed it so sorry for the delay.

- Blockquotes should be aligned with the rest of the content instead of centered:
![screenshot 2016-10-20 11 52 02](https://cloud.githubusercontent.com/assets/4924246/19573601/cd21f824-96bb-11e6-8b55-703b74ee8f84.png)

- Blockquotes color should be `grey_lighten_30` / `#e9eff3`

- In "More on WPCOM", Follow icon color is `blue_medium` / `#00aadc`

- Timestamp should say "4h ago" instead of just "4h"
![screenshot 2016-10-20 11 58 48](https://cloud.githubusercontent.com/assets/4924246/19574011/5542cafc-96bd-11e6-9097-3feb0abf3ec5.png)

- When you Follow/Unfollow from "More on WPCOM", there is a gray background on the icons that show up. Possible to get rid of it? If not, I'm happy to drop the animation: https://cloudup.com/cUqrj17zto1

- "More on [ site ]" and "More on WordPress.com" headings should be links. The first one links just within Calypso and the second links to their Following stream. On Calypso:

![screenshot 2016-10-20 12 35 30](https://cloud.githubusercontent.com/assets/4924246/19575077/bf91805c-96c1-11e6-80ba-6d88369edbcb.png)
![screenshot 2016-10-20 12 35 28](https://cloud.githubusercontent.com/assets/4924246/19575072/b86bdb24-96c1-11e6-896d-29ae27688140.png)


- If there isn't a gravatar or blavatar set, we don't show the placeholders. Instead, we drop them completely and the rest of the byline is moved to the left.

![screenshot 2016-10-20 12 13 19](https://cloud.githubusercontent.com/assets/4924246/19574461/5066b096-96bf-11e6-8f98-750fae28592c.png)

- When I come from stream > click on site header > click on post title, byline is missing: https://cloudup.com/cFvFHaY8Mki

Feel free to break this up into separate PRs if need be.

fraying commented:
Well done, Jan!

nbradbury commented:
@jancavan Ready for round two! See notes below:

> Blockquotes should be aligned with the rest of the content instead of centered...Blockquotes color should be `grey_lighten_30`

Fixed in 7adc395

> In "More on WPCOM", Follow icon color is `blue_medium`

Fixed in 0a002e3.

> Timestamp should say "4h ago" instead of just "4h"

Fixed in ede159c

> When you Follow/Unfollow from "More on WPCOM", there is a gray background on the icons that show up. Possible to get rid of it?

Fixed in 8ff1215

> "More on [ site ]" and "More on WordPress.com" headings should be links.

I mentioned this in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/4672#issuecomment-255179755).

> If there isn't a gravatar or blavatar set, we don't show the placeholders. Instead, we drop them completely and the rest of the byline is moved to the left.

Fixed in b9b8367

> When I come from stream > click on site header > click on post title, byline is missing:

Fixed in 11b2a02.

gibrown commented:
@nbradbury do the related posts here implement traintracks events and railcars per #4462 

I couldn't find them in the code, but maybe they are elsewhere. Also, based on recent Calypso experience it would be good to have the traintracks render events only get fired when the related posts get shown to the user (so when the user has scrolled that far).

Looks great.

jancavan commented:
@nbradbury Thanks for the quick turnaround! sorry I missed your comment about the hyperlink, makes sense.

I just reviewed the changes and LGTM :shipit:
